modules/secure-cloud-function-core/main.tf

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+module "cloudfunction_bucket" {
+  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
+  version = "~>3.4"
+
+  project_id    = var.project_id
+  labels        = var.labels
+  name          = "gcf-v2-sources-${var.project_number}-${var.location}"
+  location      = var.location
+  storage_class = "REGIONAL"
+  force_destroy = var.force_destroy
+
+  encryption = {
+    default_kms_key_name = var.encryption_key
+  }
+}
+
+resource "google_eventarc_google_channel_config" "primary" {
+  location        = var.location
+  name            = "projects/${var.project_id}/locations/${var.location}/googleChannelConfig"
```

Comment:
Will this cause a conflict if multiple deployments occur? If so, edit for uniqueness (function name?)

Reply:
Hey @gtsorbo 

There is only one Google channel per project, but the encryption key can be configured by location.
What can happens is to overwrite a CMEK if the user provides other key to the same location. 

It would be good to add a warning in the doc.

https://cloud.google.com/eventarc/docs/use-cmek#enable-cmek-google-channel